### PR TITLE
Add he support to pt params converter

### PR DESCRIPTION
**Pull request by SYangster**
Add tensor reshape to pt ParamsConverter to support HE for client API

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/app_opt/pt/params_converter.py [hunk 1]
```diff
@@ -14,16 +14,22 @@
...
+        tensor_shapes = fl_ctx.get_prop("tensor_shape_dict")
+        if tensor_shapes:
+            return {k: torch.as_tensor(np.reshape(v, tensor_shapes[k])) for k, v in params.items()}
+        else:
+            return {k: torch.as_tensor(v) for k, v in params.items()}
 
 
 class PTToNumpyParamsConverter(ParamsConverter):
     def convert(self, params: Dict, fl_ctx) -> Dict:
+        fl_ctx.set_prop("tensor_shape_dict", {k: v.shape for k, v in params.items()})
```
[the remark this answers is not in the document]
Comment (holgerroth) [reply nvflare/app_opt/pt/params_converter.py:34]: Yes, only the shape is needed. I think this `v.shape` is only saving the shapes.

File: nvflare/app_opt/pt/params_converter.py [hunk 2]
```diff
@@ -14,16 +14,22 @@
...
 
 from nvflare.app_common.abstract.params_converter import ParamsConverter
 
 
 class NumpyToPTParamsConverter(ParamsConverter):
     def convert(self, params: Dict, fl_ctx) -> Dict:
-        return {k: torch.as_tensor(v) for k, v in params.items()}
+        tensor_shapes = fl_ctx.get_prop("tensor_shape_dict")
+        if tensor_shapes:
+            return {k: torch.as_tensor(np.reshape(v, tensor_shapes[k])) for k, v in params.items()}
```
Comment (holgerroth) [inline nvflare/app_opt/pt/params_converter.py:27]: should we only reshape if `k` is in ` tensor_shapes`?

Comment (SYangster) [reply nvflare/app_opt/pt/params_converter.py:27]: sure, good idea

File: job_templates/sag_pt_he/config_fed_server.conf [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      args {
+        model {
+          path = "{model_class_path}"
+        }
+        filter_id = "serialize_filter"
+      }
+    },
+    {
+      id = "shareable_generator"
+      name = "HEModelShareableGenerator"
```
Comment (holgerroth) [inline job_templates/sag_pt_he/config_fed_server.conf:85]: Should we import by path?

Comment (SYangster) [reply job_templates/sag_pt_he/config_fed_server.conf]: changed

**Review by holgerroth — APPROVED** [review]
Looks good. Runs fine on my side.

**Pull request merged**